lib/Doctrine/ORM/Query.php

```diff
@@ -147,6 +149,9 @@ class Query extends AbstractQuery
      */
     private $parserResult;
 
+    /** @var AbstractSqlExecutor */
```

Comment:
Why not allow `null` here since you're assigning `null` to that property?

```suggestion
    /** @var AbstractSqlExecutor|null */
```